pkg/controller/service/controller_test.go

```diff
@@ -499,6 +499,52 @@ func TestUpdateNodesInExternalLoadBalancer(t *testing.T) {
 	}
 }
 
+func TestGetNodeConditionPredicate(t *testing.T) {
+	controller, _, _ := newController()
+
+	tests := []struct {
+		node         v1.Node
+		expectAccept bool
+		name         string
+	}{
+		{
+			node:         v1.Node{},
+			expectAccept: false,
+			name:         "nodes that are not ready are not included",
+		},
+		{
+			node: v1.Node{
+				Status: v1.NodeStatus{
+					Conditions: []v1.NodeCondition{
+						{Type: v1.NodeReady, Status: v1.ConditionTrue},
+					},
+				},
+			},
+			expectAccept: true,
+			name:         "healthy nodes are included",
+		},
+		{
+			node: v1.Node{
+				Spec: v1.NodeSpec{Unschedulable: true},
+				Status: v1.NodeStatus{
+					Conditions: []v1.NodeCondition{
+						{Type: v1.NodeReady, Status: v1.ConditionTrue},
+					},
+				},
+			},
+			expectAccept: true,
+			name:         "unschedulable but healthy nodes are included",
+		},
+	}
+	pred := controller.getNodeConditionPredicate()
```

Comment:
I was going to suggest this also test the legacy label and the node-role label, but then I thought, "I swear there's a test already".

Why not merge this test func and the existing `Test_getNodeConditionPredicate()` ?

Reply:
I can do that.